File: virtual-classroom/classroom/models/course.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace classroom.models
{
    
        public class course
        {
            [Key]
             
            public int id { get; set; }

        
            public string name { get; set; }
 
            
          
            public string description { get; set; }

             
            public int Material_id { get; set; }

            

             
            public int teacher_id { get; set; }

        
            public string type { get; set; }
   
            public DateTime date { get; set; }

        
            public TimeSpan start_time { get; set; }

 

        public int Student_id { get; set; }
        }
       
    
}

File: virtual-classroom/frontend/src/shared/Model/course.ts
import { Time } from '@angular/common';

export interface Course {
  id: number;
  name: string;
  description: string;
  teacher_id: number;
  material_id: number;
  start_time: string;
  student_id: number;
  type: string;
  date: Date;
  time: Time;
}
